includes/classes/Indexable/Post/Post.php

```diff
@@ -45,6 +45,8 @@ public function __construct() {
 
 		$this->sync_manager      = new SyncManager( $this->slug );
 		$this->query_integration = new QueryIntegration( $this->slug );
+
+		add_filter( 'posts_where', array( $this, 'bulk_indexing_filter_posts_where' ), 9999, 2 );
```

Comment:
I agree it's not super great, but also can't find a better option :/. 

- `Command->index_helper()` is technically when the bulk index query is kicked off and where it finishes, though this is object agnostic and it would be even more "off" doing this there.
- Could check if the filter needs to be added during `Indexable\Post->query_db()` perhaps and do so before the query runs, though that's logic that will run a lot more than where the filter currently sites.

What are you thinking would be better than how it currently is?

Reply:
Gonna go with option 2. This is indeed a problem during tests, was ending up with:

```
AND wptests_posts.ID <= 9 AND wptests_posts.ID >= 0 AND wptests_posts.ID <= 9 AND wptests_posts.ID >= 0
```